### PR TITLE
Update channel open to be async and for new upstream API

### PR DESCRIPTION
**Pull request by TheBlueMatt**

### CODE REVIEW

File: docs/build_node.md [hunk 1]
```diff
@@ -381,7 +381,60 @@ This section assumes you've followed the steps of the [Setup](#setup) and [Runni
...
+    peer_node_pubkey, 10000, 1000, 42, null);
+assert create_channel_result instanceof Result_NoneAPIErrorZ.Result_NoneAPIErrorZ_OK;
+
+// Ensure we immediately send a `create_channel` message to the counterparty.
+nio_peer_handler.check_events();
+
+// After the peer responds with an `accept_channel` message, an Event.FundingGenerationReady
+// event will be generated.
+
+// In the background event handler thread (see section above on "Handle...Generated Events"),
```
Comment (valentinewallace) [inline docs/build_node.md:401]: Could you rename the event mentioned^ to "Handle LDK Events" and rename that here :grimacing:

File: docs/build_node.md [hunk 2]
```diff
@@ -381,7 +381,60 @@ This section assumes you've followed the steps of the [Setup](#setup) and [Runni
...
+	NetworkParameters bitcoinj_net = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+	Transaction funding_tx = new Transaction(bitcoinj_net);
+	funding_tx.addInput(new TransactionInput(bitcoinj_net, funding, new byte[0]));
+	// Note that all inputs in the funding transaction MUST spend SegWit outputs (and have witnesses)
+	funding_tx.getInputs().get(0).setWitness(new TransactionWitness(2));
+	funding_tx.getInput(0).getWitness().setPush(0, new byte[]{0x1});
+	funding_tx.addOutput(Coin.SATOSHI.multiply(output_value), new Script(funding_scriptpubkey));
+	short funding_output_index = (short) 0;
+
+	// Give the funding transaction back to the channel manager.
```
Comment (valentinewallace) [inline docs/build_node.md:421]: nit: s/channel manager/ChannelManager

File: docs/build_node.md [hunk 3]
```diff
@@ -381,7 +381,60 @@ This section assumes you've followed the steps of the [Setup](#setup) and [Runni
...
+	// funding_transaction_generated should only generate an error if the transaction didn't meet
+	// the required format (or the counterparty already closed the channel on us):
+    assert funding_res instanceof Result_NoneAPIErrorZ.Result_NoneAPIErrorZ_OK;
+
+	// Ensure we immediately send a `funding_created` message to the counterparty.
+	nio_peer_handler.check_events();
+
+	// At this point LDK will exchange the remaining channel open messages with the counterparty and,
+	// when appropriate, broadcast the funding transaction provided.
+	// Once it confirms, the channel will be opan and available for use (indicated by its presence in
```
Comment (valentinewallace) [inline docs/build_node.md:434]: nit: s/opan/open

**Review by valentinewallace — COMMENTED** [review]
The code looks sane! A few nits basically

**Comment by TheBlueMatt** [comment]
Thanks! Addressed comments in fixup commits, let me know if you want me to squash.

File: docs/build_node.md [hunk 4]
```diff
@@ -381,7 +381,70 @@ This section assumes you've followed the steps of the [Setup](#setup) and [Runni
...
+	byte[] funding_scriptpubkey = ((Event.FundingGenerationReady) e).output_script;
+	long output_value = ((Event.FundingGenerationReady) e).channel_value_satoshis;
+	// This is the same channel we just created, above:
+	assert ((Event.FundingGenerationReady) e).user_channel_id == 42;
+	// The output is always a P2WSH:
+	assert funding_scriptpubkey.length == 34 && funding_scriptpubkey[0] == 0 &&
+		funding_scriptpubkey[1] == 32;
+
+	// Generate the funding transaction for the channel based on the channel amount
+	NetworkParameters bitcoinj_net = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
```
Comment (valentinewallace) [inline docs/build_node.md:417]: 80chars plz

**Comment by valentinewallace** [comment]
> Thanks! Addressed comments in fixup commits, let me know if you want me to squash.

Hmm.. Squashing is slightly preferred but nbd.

**Comment by TheBlueMatt** [comment]
Addressed and squashed.

**Pull request merged**